src/utilities/graphql/fragments.ts

```diff
@@ -143,3 +144,26 @@ export function getFragmentFromSelection(
       return null;
   }
 }
+
+export function isFullyUnmaskedOperation(document: DocumentNode) {
+  let isUnmasked = true;
+
+  visit(document, {
+    FragmentSpread: (node) => {
+      if (!node.directives) {
+        isUnmasked = false;
+        return BREAK;
+      }
+
+      isUnmasked &&= node.directives.some(
+        (directive) => directive.name.value === "unmask"
+      );
+
+      if (!isUnmasked) {
+        return BREAK;
+      }
+    },
```

Comment:
```suggestion
    FragmentSpread: (node) => {
      isUnmasked = !!node.directives && node.directives.some(
        (directive) => directive.name.value === "unmask"
      );

      if (!isUnmasked) {
        return BREAK;
      }
    },
```

I think this whole block could be simplified like this.